### ISSUE TITLE
Implement Offline Support

### ISSUE THREAD

**Issue by kmahelona** [TeHikuMedia/corpora-expo#7]
## Offline support ##

We need the ability to record sentences even when there isn't an internet connection (or the connection is intermittent). The following views should support this. They are ordered with highest priority first.

### 1. Reading View ###

1. Prefetch X number of sentences for reading purposes and store locally.
2. Record View takes a sentence from the local sentence storage, records a user reading the sentences, then stores the recording and associated meta data locally
3. Queue the local recording object for uploading to the API
4. When the app is active (we don't need background support) continuously check for queued recordings and upload them to the API when there's an internet connection.

### 2. Listen View ###

1. Prefect X number of recordings. We'll need to thing about storage space etc.
2. Listen view takes a recording from the local listen storage, plays it to the user, then the user votes on whether the recording is good or bad, and then store the user's vote locally.
3. Queue the users vote for uploading to the API
4. When the app is active (we don't need background support) continuously check for queued votes and upload them to the API when there's an internet connection.

**Comment by kmahelona** [comment]
Hey @richchurcher just making a git issue for the offline support. Please feel free to modify my suggestions above. Sounds like you've already got it sorted.

**Comment by kmahelona** [comment]
This seems to be working well. 

I noticed that my recordings weren't being saved in this scenario,

1. Latest version installed (https://github.com/TeHikuMedia/corpora-expo/releases/tag/v1.0.2)
2. Had app open since yesterday
3. Did about 5 recordings and they didn't upload
4. Restarted the app
5. The recordings uploaded.

I did have an internet connection. So somewhere in there it must have thought I was offline? It's great the the recordings made it!

This should be investigated. We need to ensure to upload all data as soon as possible. Users will get very 😡if they did work and realised their work wasn't save. We learned this one the hard way. I'll note this as done and open a new issue.
